src/logger.cpp

```diff
@@ -0,0 +1,31 @@
+#include "pgduckdb/logger.hpp"
+
+extern "C" {
+#include "postgres.h"
+}
+
+#include <utility>
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wbuiltin-macro-redefined"
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wbuiltin-macro-redefined"
+
+// Redefine them to match the pgduckdb logger so it uses the provided filename and lineno
+#define __FILE__ filename
+#define __LINE__ lineno
```

Comment:
This is not ideal but the only way I could find to wrap the `elog` macro while preserving original file name & line.